sematic/future_context.py

```diff
@@ -0,0 +1,88 @@
+# Standard Library
+from contextlib import contextmanager
+from dataclasses import dataclass
+from importlib import import_module
+from typing import Any, Optional, Type
+
+
+@dataclass(frozen=True)
+class SematicContext:
+    """Contextual information about the execution of the current Sematic function
+    Attributes
+    ----------
+    id:
+        The id of the future for the current execution. For cloud executions, this
+        is equivalent to the id for the existing run.
+    root_id:
+        The id of the root future for a resolution. For cloud executions, this is
+        equivalent to the id for the root run.
+    resolver_class_path:
+        The import path for the resolver being used.
+    """
+
+    id: str
+    root_id: str
+    resolver_class_path: str
+
+    def resolver_class(self) -> Type[Any]:
```

Comment:
Does this work?
```suggestion
    def resolver_class(self) -> Type[Resolver]:
```

Reply:
I was originally thinking it would create a circular reference (resolvers <-> future context). However, it appears `Resolver` is just an abstract base with basically nothing in it so using that will be fine